tests/fs_after_main.cpp

```diff
@@ -38,29 +38,51 @@ int main() {
     (function() {
       // exiting main should not cause any weirdness with printing
       var realPrint = Module['print'];
+      var realPrintChars = Module['printChars'];
       Module['extraSecretBuffer'] = '';
       Module['print'] = function(x) {
         Module['extraSecretBuffer'] += x;
         realPrint(x);
       };
+      Module['printChars'] = function(str_or_offset, fd, len, buffer) {
+        var str = UTF8ArrayToString((typeof str_or_offset === 'number') ? buffer : str_or_offset, str_or_offset, len);
+        Module['extraSecretBuffer'] += str;
+        realPrintChars(str_or_offset, fd, len, buffer);
+      }
     })();
   });
   printf("Start\n");
   FILE* f = fopen(NAME, "wb");
   fclose(f);
   printf("Looping...\n");
   EM_ASM({
+#if USE_PRINTCHARS_FOR_STDOUT == 0
```

Comment:
These would be nice to also read `#if USE_PRINTCHARS_FOR_STDOUT` to read in the positive direction.